core/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -739,6 +745,14 @@ public Builder put(Object... settings) {
          * @return The builder
          */
         public Builder put(String key, String value) {
+            if (key.endsWith(".0")) {
```

Comment:
This makes me uncomfortable.... I see that we used to do it as well. Can we get away with not doing it anymore now that we have `putArray`?